src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -36,16 +37,59 @@
 
     public static final String MESSAGE_INVALID_INDEX = "Index is not a non-zero unsigned integer.";
 
+    public static final String MESSAGE_INDEX_OUT_OF_BOUNDS = "The index provided is invalid";
+
+    /**
+     * Returns true if string has length less than or equal to 9 after trimming leading zeroes.
+     *
+     * @param trimmedIndex trimmedIndex
+     * @return whether index is within the acceptable range
+     */
+    private static boolean isWithinIndexRange(String trimmedIndex) {
+        int currIndex = 0;
+        for (int i = 0; i < trimmedIndex.length(); i++) {
+            char c = trimmedIndex.charAt(i);
+            if (c != '0') {
+                break;
+            } else {
+                currIndex += 1;
+            }
+        }
+        String removedLeadingZeros = trimmedIndex.substring(currIndex);
+        return !(removedLeadingZeros.length() > 9);
+    }
+
+    /**
+     * Returns true is string contains all digits from 0 to 9
+     * @param s
+     * @return
+     */
+    private static boolean isAllDigits(String s) {
```

Comment:
I think a regex check suffices here. `return s.matches("^\\d+$")`

Reply:
Thanks for the suggestion! Have replaced the existing function with this regex.